pandas/tests/io/json/test_normalize.py

```diff
@@ -653,16 +663,44 @@ def test_missing_nested_meta(self):
         )
         tm.assert_frame_equal(result, expected)
 
-        # If errors="raise" and nested metadata is null, we should raise with the
-        # key of the first missing level
+        # If errors="raise" and nested metadata is null,
+        # should NOT raise if the missing value is leaf value
+        result = json_normalize(
+            data,
+            record_path="value",
+            meta=["meta", ["nested_meta", "leaf"]],
+            errors="raise",
+        )
+        tm.assert_frame_equal(result, expected)
+
+        # If errors="raise" and nested metadata is null,
+        # should raise if the missing value is node value:
+        # see ["nested_meta", "leaf", "leaf"] instruction
+
         with pytest.raises(KeyError, match="'leaf' not found"):
             json_normalize(
                 data,
                 record_path="value",
-                meta=["meta", ["nested_meta", "leaf"]],
+                meta=["meta", ["nested_meta", "leaf", "leaf"]],
```

Comment:
I don't quite understand why we needed to add another level to get this to fail - there is no `nested_meta.leaf` path to begin with so shouldn't that fail on its own?